gremlin-javascript/src/main/javascript/gremlin-javascript/test/integration/socket-connection-tests.js

```diff
@@ -49,7 +49,49 @@ describe('Connection', function () {
   });
 
   describe('#open()', function () {
+    it('should use the ws WebSocket when ws specific options are provided', function () {
+      let globalWebsocketCalls = 0;
+      globalThis.WebSocket = function () {
+        globalWebsocketCalls++;
+      };
+      const wsSpecificOptions = [
+        'headers',
+        'ca',
+        'cert',
+        'pfx',
+        'rejectUnauthorized',
+        'agent',
+        'perMessageDeflate',
+      ];
+      const allOptionTests = wsSpecificOptions.map((wsOption) => {
+        const connection = helper.getDriverRemoteConnection(`ws://localhost:${testServerPort}/401`, {
+          [wsOption]: 'this option is set',
+        });
+        return connection.open();
+      });
+      Promise.allSettled(allOptionTests).then(function () {
+        if (globalWebsocketCalls !== 0) {
+          assert.fail('global WebSocket should be used when no ws specific options are provided');
+        }
```

Comment:
This may be subjective, but could this `if` statement be an assertion instead?

```js
assert.equal(globalWebsocketCalls, 0, 'global WebSocket should be used when no ws specific options are provided');
```

Reply:
Pushed the change for this!